rpcserver.go

```diff
@@ -7738,77 +7726,72 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 			return err
 		}
 
-		resp, err := r.AddAssetSellOrder(
-			ctx, &rfqrpc.AddAssetSellOrderRequest{
-				AssetSpecifier: &rpcSpecifier,
-				PaymentMaxAmt:  uint64(paymentMaxAmt),
-				Expiry:         uint64(expiry.Unix()),
-				PeerPubKey:     peerPubKey[:],
-				TimeoutSeconds: uint32(
-					rfq.DefaultTimeout.Seconds(),
-				),
+		rpcCtx, _, client := r.cfg.Lnd.Client.RawClientWithMacAuth(ctx)
+		payReqInfo, err := client.DecodePayReq(
+			rpcCtx, &lnrpc.PayReqString{
+				PayReq: req.PaymentRequest.PaymentRequest,
 			},
 		)
 		if err != nil {
-			return fmt.Errorf("error adding sell order: %w", err)
+			return err
 		}
 
-		var acceptedQuote *rfqrpc.PeerAcceptedSellQuote
-		switch r := resp.Response.(type) {
-		case *rfqrpc.AddAssetSellOrderResponse_AcceptedQuote:
-			acceptedQuote = r.AcceptedQuote
+		payHash := payReqInfo.PaymentHash
 
-		case *rfqrpc.AddAssetSellOrderResponse_InvalidQuote:
-			return fmt.Errorf("peer %v sent back an invalid "+
-				"quote, status: %v", r.InvalidQuote.Peer,
-				r.InvalidQuote.Status.String())
+		// Since we don't have any pre-negotiated rfq IDs available, we
+		// need to initiate the negotiation procedure for this payment.
+		// We'll store here all the quotes we acquired successfully.
+		acquiredQuotes := r.acquirePaymentQuotes(
+			ctx, &rpcSpecifier, paymentMaxAmt, expiry, chanMap,
+			req.AllowOverpay, payHash,
+		)
 
-		case *rfqrpc.AddAssetSellOrderResponse_RejectedQuote:
-			return fmt.Errorf("peer %v rejected the quote, code: "+
-				"%v, error message: %v", r.RejectedQuote.Peer,
-				r.RejectedQuote.ErrorCode,
-				r.RejectedQuote.ErrorMessage)
+		// Send out the information about the acquired quotes on the
+		// stream.
+		for _, quote := range acquiredQuotes {
```

Comment:
This could also be extracted into a function, would make it a bit more explicit that this only informs the user and that call to `rpcutils.UnmarshalRfqFixedPoin()` is only for the log statement.